apps/distgo/params/docker.go

```diff
@@ -60,7 +68,82 @@ type DockerImage struct {
 	// This will be used to order the dist tasks such that all the dependent
 	// products' dist tasks will be executed first, after which the dist tasks for the
 	// current product are executed.
-	Deps DockerDeps
+	Deps []DockerDep
+}
+
+type SLSDockerImage struct {
+	DefaultDockerImage
+	GroupID      string
+	ProuductType string
+	Extensions   map[string]interface{}
+}
+
+func (sdi *SLSDockerImage) Build(buildSpec ProductBuildSpecWithDeps) error {
+	contextDir := path.Join(buildSpec.Spec.ProjectDir, sdi.ContextDir)
+	configFile := path.Join(contextDir, ConfigurationFileName)
+	var args []string
+	args = append(args, "build")
+	args = append(args, "--tag", fmt.Sprintf("%s:%s", sdi.Repository, sdi.Tag))
+	manifest, err := GetManifest(sdi.GroupID, buildSpec.Spec.ProductName, buildSpec.Spec.ProductVersion, sdi.ProuductType, sdi.Extensions)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get manifest for the image")
+	}
+	args = append(args, "--label", fmt.Sprintf("%s=%s", ManifestLabel, base64.StdEncoding.EncodeToString([]byte(manifest))))
+	if _, err := os.Stat(configFile); err == nil {
+		content, err := ioutil.ReadFile(configFile)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to read the file %s", configFile)
+		}
+		args = append(args, "--label", fmt.Sprintf("%s=%s", ConfigurationLabel, base64.StdEncoding.EncodeToString(content)))
+	} else {
+		fmt.Printf("Unable to find the configuration.yml file at: %v\n", configFile)
```

Comment:
Shouldn't be using `Printf` here -- use `Fprintf` to a `stdout` stream instead.

Also, is this an error case where you should return an error? I would think that either it's an error (in which case you should return an error) or OK (in which case you can just proceed without printing any output)